src/Catel.Core/IoC/EventArgs/TypeInstantiatedEventArgs.cs

```diff
@@ -21,9 +20,10 @@ public class TypeInstantiatedEventArgs : EventArgs
         /// <param name="serviceImplementationType">Type of the service implementation.</param>
         /// <param name="tag">The tag.</param>
         /// <param name="registrationType">Type of the registration.</param>
+        /// <param name="instance">The instance of a service.</param>
         /// <exception cref="ArgumentNullException">The <paramref name="serviceType"/> is <c>null</c>.</exception>
         /// <exception cref="ArgumentNullException">The <paramref name="serviceImplementationType"/> is <c>null</c>.</exception>
-        public TypeInstantiatedEventArgs(Type serviceType, Type serviceImplementationType, object tag, RegistrationType registrationType)
+        public TypeInstantiatedEventArgs(Type serviceType, Type serviceImplementationType, object tag, RegistrationType registrationType, object instance)
```

Comment:
This is a breaking change, we need to be careful introducing new breaking changes. I think this one is "safe" since it will only be instantiated by Catel itself.

Reply:
I also thought that it's safe, but there might be some custom implementations of IServiceLocator, so I'll make another constructor.